src/values/ObjectValue.js

```diff
@@ -585,8 +585,13 @@ export default class ObjectValue extends ConcreteValue {
       !this.isSimpleObject() ||
       (P.mightNotBeString() && P.mightNotBeNumber() && !P.isSimpleObject())
     ) {
-      AbstractValue.reportIntrospectionError(P, "TODO: #1021");
-      throw new FatalError();
+      // if P is an abstract value that we don't know about, but we're in pure scope
+      // then if the object is simple, then we can safely continue without throwing
+      // the below introspection error
```

Comment:
the below introspection error -> the introspection error below, since converting P to a string is assumed to be well behaved in a pure scope